Cannot read properties of null (reading 'isDisabled')

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.6.3

### Please select which platform you are using if self-hosting.

Azure DevOps (dev.azure.com)

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

It seems as if Renovate is unable to fetch the list of repositories of my organization/project.

These are the permissions of the build user
![image](https://user-images.githubusercontent.com/7439596/159694628-5aa4fab8-deb5-4a70-85dc-0eab9f403518.png)



### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Using localDir: /tmp/renovate/repos/azure/CCA/Hub (repository=CCA/Hub)
DEBUG: initRepo("CCA/Hub") (repository=CCA/Hub)
TRACE: getRepoByName(CCA/Hub) (repository=CCA/Hub)
TRACE: getProjectAndRepo(CCA/Hub) (repository=CCA/Hub)
DEBUG: Repository details (repository=CCA/Hub)
       "repositoryDetails": null
ERROR: Repository has unknown error (repository=CCA/Hub)
       "err": {
         "message": "Cannot read properties of null (reading 'isDisabled')",
         "stack": "TypeError: Cannot read properties of null (reading 'isDisabled')\n    at Proxy.initRepo (/home/vsts/.npm/_npx/05eeecd92f4e18e0/node_modules/renovate/dist/modules/platform/azure/index.js:87:14)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async getPlatformConfig (/home/vsts/.npm/_npx/05eeecd92f4e18e0/node_modules/renovate/dist/workers/repository/init/apis.js:38:28)\n    at async initApis (/home/vsts/.npm/_npx/05eeecd92f4e18e0/node_modules/renovate/dist/workers/repository/init/apis.js:47:14)\n    at async initRepo (/home/vsts/.npm/_npx/05eeecd92f4e18e0/node_modules/renovate/dist/workers/repository/init/index.js:25:14)\n    at async Object.renovateRepository (/home/vsts/.npm/_npx/05eeecd92f4e18e0/node_modules/renovate/dist/workers/repository/index.js:37:18)\n    at async Object.start (/home/vsts/.npm/_npx/05eeecd92f4e18e0/node_modules/renovate/dist/workers/global/index.js:116:13)\n    at async /home/vsts/.npm/_npx/05eeecd92f4e18e0/node_modules/renovate/dist/renovate.js:15:24"
       }
DEBUG: Unknown res (repository=CCA/Hub)
       "res": "unknown-error"
DEBUG: Repository timing splits (milliseconds) (repository=CCA/Hub)
       "splits": {},
       "total": 314
 INFO: Repository finished (repository=CCA/Hub)
       "durationMs": 314
DEBUG: Renovate exiting
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 50,
           "logContext": "8cPsjaDErrk44gBGyZ7IR",
           "repository": "CCA/Hub",
           "err": {
             "message": "Cannot read properties of null (reading 'isDisabled')",
             "stack": "TypeError: Cannot read properties of null (reading 'isDisabled')\n    at Proxy.initRepo (/home/vsts/.npm/_npx/05eeecd92f4e18e0/node_modules/renovate/dist/modules/platform/azure/index.js:87:14)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async getPlatformConfig (/home/vsts/.npm/_npx/05eeecd92f4e18e0/node_modules/renovate/dist/workers/repository/init/apis.js:38:28)\n    at async initApis (/home/vsts/.npm/_npx/05eeecd92f4e18e0/node_modules/renovate/dist/workers/repository/init/apis.js:47:14)\n    at async initRepo (/home/vsts/.npm/_npx/05eeecd92f4e18e0/node_modules/renovate/dist/workers/repository/init/index.js:25:14)\n    at async Object.renovateRepository (/home/vsts/.npm/_npx/05eeecd92f4e18e0/node_modules/renovate/dist/workers/repository/index.js:37:18)\n    at async Object.start (/home/vsts/.npm/_npx/05eeecd92f4e18e0/node_modules/renovate/dist/workers/global/index.js:116:13)\n    at async /home/vsts/.npm/_npx/05eeecd92f4e18e0/node_modules/renovate/dist/renovate.js:15:24"
           },
           "msg": "Repository has unknown error"
         }
       ]
##[error]Bash exited with code '1'.

```

</details>


### Have you created a minimal reproduction repository?

No reproduction repository

--- a/lib/modules/platform/azure/index.spec.ts
+++ b/lib/modules/platform/azure/index.spec.ts
@@ -5,7 +5,10 @@ import {
   GitStatusState,
   PullRequestStatus,
 } from 'azure-devops-node-api/interfaces/GitInterfaces.js';
-import { REPOSITORY_ARCHIVED } from '../../../constants/error-messages';
+import {
+  REPOSITORY_ARCHIVED,
+  REPOSITORY_NOT_FOUND,
+} from '../../../constants/error-messages';
 import type { logger as _logger } from '../../../logger';
 import { BranchStatus, PrState } from '../../../types';
 import type * as _git from '../../../util/git';
@@ -183,6 +186,14 @@ describe('modules/platform/azure/index', () => {
           repository: 'some/repo3',
         })
       ).rejects.toThrow(REPOSITORY_ARCHIVED);
+    });
+
+    it(`throws if repo is not in repos list`, async () => {
+      await expect(
+        initRepo({
+          repository: 'some/missing',
+        })
+      ).rejects.toThrow(REPOSITORY_NOT_FOUND);
     });
   });
 

--- a/lib/modules/platform/azure/index.ts
+++ b/lib/modules/platform/azure/index.ts
@@ -173,7 +173,6 @@ export async function initRepo({
   const azureApiGit = await azureApi.gitApi();
   const repos = await azureApiGit.getRepositories();
   const repo = getRepoByName(repository, repos);
-  // istanbul ignore if
   if (!repo) {
     logger.error({ repos, repo }, 'Could not find repo in repo list');
     throw new Error(REPOSITORY_NOT_FOUND);

--- a/lib/modules/platform/azure/index.ts
+++ b/lib/modules/platform/azure/index.ts
@@ -14,6 +14,7 @@ import { PlatformId } from '../../../constants';
 import {
   REPOSITORY_ARCHIVED,
   REPOSITORY_EMPTY,
+  REPOSITORY_NOT_FOUND,
 } from '../../../constants/error-messages';
 import { logger } from '../../../logger';
 import { BranchStatus, PrState, VulnerabilityAlert } from '../../../types';
@@ -172,6 +173,11 @@ export async function initRepo({
   const azureApiGit = await azureApi.gitApi();
   const repos = await azureApiGit.getRepositories();
   const repo = getRepoByName(repository, repos);
+  // istanbul ignore if
+  if (!repo) {
+    logger.error({ repos, repo }, 'Could not find repo in repo list');
+    throw new Error(REPOSITORY_NOT_FOUND);
+  }
   logger.debug({ repositoryDetails: repo }, 'Repository details');
   if (repo.isDisabled) {
     logger.debug('Repository is disabled- throwing error to abort renovation');

